src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -1718,54 +1718,39 @@ SPECTRE_ALWAYS_INLINE constexpr auto mutate_apply(
   return mutate_apply(F{}, box, std::forward<Args>(args)...);
 }
 
-// @{
-/*!
- * \ingroup DataBoxGroup
- * \brief Perform a mutation to the \ref DataBoxGroup `box`, assigning the
- * `args` to the tags `MutateTags` in order.
- */
+namespace DataBox_detail {
 template <typename... MutateTags, typename BoxTags, typename... Args>
-SPECTRE_ALWAYS_INLINE constexpr void mutate_assign(
+SPECTRE_ALWAYS_INLINE constexpr void mutate_assign_impl(
     // NOLINTNEXTLINE(readability-avoid-const-params-in-decls)
-    const gsl::not_null<DataBox<BoxTags>*> box, Args&&... args) noexcept {
+    const gsl::not_null<DataBox<BoxTags>*> box,
+    tmpl::list<MutateTags...> /*meta*/, Args&&... args) noexcept {
   static_assert(sizeof...(MutateTags) == sizeof...(args),
                 "The number of arguments passed to `mutate_assign` must be "
                 "equal to the number of tags passed.");
   const auto assign = [](const auto lhs, auto&& rhs) noexcept {
-    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
     *lhs = std::forward<decltype(rhs)>(rhs);
   };
   db::mutate<MutateTags...>(
-      box,
-      // NOLINTNEXTLINE(modernize-avoid-c-arrays)
-      [&assign, &args...](const auto... box_args) noexcept {
+      box, [&assign, &args...](const auto... box_args) noexcept {
         // silence unused capture warnings
         (void)assign;
         EXPAND_PACK_LEFT_TO_RIGHT((void)args);
         EXPAND_PACK_LEFT_TO_RIGHT(assign(box_args, std::forward<Args>(args)));
```

Comment:
I think you can do:

```cpp
expand_pack((*box_args = std::forward<Args>(args))...);
```

Reply:
Ah, yeah, I noticed that on your PR. I'll switch to that, though I think I'll keep the order preservation because I believe databox assignment can actually depend on order sometimes (e.g. subtags).